src/store.ts

```diff
@@ -1025,7 +1064,96 @@ const createMlStore = (logging: Logging) => {
               set({ predictionInterval: undefined });
             }
           },
+
+          languageDialogOnOpen() {
+            set({ isLanguageDialogOpen: true });
+          },
+          settingsDialogOnOpen() {
+            set({ isSettingsDialogOpen: true });
+          },
+          connectFirstDialogOnOpen() {
+            set({ isConnectFirstDialogOpen: true });
+          },
+          aboutDialogOnOpen() {
+            set({ isAboutDialogOpen: true });
+          },
+          feedbackFormOnOpen() {
+            set({ isFeedbackFormOpen: true });
+          },
+          deleteAllActionsDialogOnOpen() {
+            set({ isDeleteAllActionsDialogOpen: true });
+          },
+          nameProjectDialogOnOpen() {
+            set({ isNameProjectDialogOpen: true });
+          },
+          recordingDialogOnOpen() {
+            set({ isRecordingDialogOpen: true });
+          },
+          connectToRecordDialogOnOpen() {
+            set({ isConnectToRecordDialogOpen: true });
+          },
+          deleteActionDialogOnOpen() {
+            set({ isDeleteActionDialogOpen: true });
+          },
+          incompatibleEditorDeviceDialogOnOpen() {
+            set({ isIncompatibleEditorDeviceDialogOpen: true });
+          },
+          closeDialog() {
+            set({
+              isLanguageDialogOpen: false,
+              isSettingsDialogOpen: false,
+              isConnectFirstDialogOpen: false,
+              isAboutDialogOpen: false,
+              isFeedbackFormOpen: false,
+              isDeleteAllActionsDialogOpen: false,
+              isNameProjectDialogOpen: false,
+              isRecordingDialogOpen: false,
+              isConnectToRecordDialogOpen: false,
+              isDeleteActionDialogOpen: false,
+              isIncompatibleEditorDeviceDialogOpen: false,
+            });
+          },
+
+          isNonConnectionDialogOpen() {
```

Comment:
Terrible name, suggestions welcomed. This function returns `true` if any dialog that is not a connection flow dialog is opened, otherwise it returns `false`. I ignore the state of a connection flow dialog because in `DefaultPageLayout.tsx` I use this function to suppress connection flow dialog if any other dialog is opened.

Reply:
Hoping that eventually the state for that moves in here and we can tidy it then.